libs/application-generic/src/usecases/upsert-preferences/upsert-preferences.usecase.ts

```diff
@@ -61,12 +61,12 @@ export class UpsertPreferences {
     });
   }
 
-  private async upsert(
+  public async upsert(
     command: UpsertPreferencesCommand,
   ): Promise<PreferencesEntity> {
     const foundId = await this.getPreferencesId(command);
 
-    if (command.preferences === null) {
+    if (command.preferences === null && foundId) {
       return this.deletePreferences(command, foundId);
     }
 
```

Comment:
`this.deletePreferences` needs `foundId` for deletion. If the `foundId` is undefined, it throws and error.

Reply:
You're spot on with the `foundId` part, I made the same change in my PR.

Reply:
great mind think alike 😉